components/Ink/src/MDCInkView.m

```diff
@@ -103,30 +122,69 @@ - (void)setCustomInkCenter:(CGPoint)customInkCenter {
   self.inkLayer.customInkCenter = customInkCenter;
 }
 
-- (MDCInkLayer *)inkLayer {
-  return (MDCInkLayer *)self.layer;
+- (MDCLegacyInkLayer *)inkLayer {
+  return (MDCLegacyInkLayer *)self.layer;
 }
 
 - (void)startTouchBeganAnimationAtPoint:(CGPoint)point
                              completion:(MDCInkCompletionBlock)completionBlock {
-  [self.inkLayer spreadFromPoint:point completion:completionBlock];
+  if (self.usesLegacyInkRipple) {
+    [self.inkLayer spreadFromPoint:point completion:completionBlock];
+  } else {
+    self.startInkRippleCompletionBlock = completionBlock;
```

Comment:
It looks like we might be losing prior completion blocks here - is that intentional?

Reply:
The intent is to only have one completion block and to only fire one completion block, rather than have a completion block fire for each ink ripple when the user taps quickly. The completion block should be associated with the latest ripple.

Reply:
Ah, I see. I'll file a bug so we can discuss improving the clarity of the completion handling APIs then.

https://github.com/material-components/material-components-ios/issues/2520

Reply:
Sounds good. Will discuss further on the bug. For this case, having a protocol on MDCInkView that is similar to MDCInkLayerDelegate might be more appropriate than using completion blocks. For the time being, I'm following the status quo that's in place for legacy ink.